### PR TITLE
stream: add StreamExt::skip

### PR DESCRIPTION
**Pull request by torepettersen**
This PR introduces StreamExt::skip as described in #2104. Skip will create a new stream the skips the n first elements of the underlaying stream.

### CODE REVIEW

File: tokio/src/stream/mod.rs [hunk 1]
```diff
@@ -451,6 +454,31 @@ pub trait StreamExt: Stream {
         TakeWhile::new(self, f)
     }
 
+    /// Creates a new stream that will skip the `n` first items of the
+    /// undelaying stream.
```
Comment (Darksonn) [inline tokio/src/stream/mod.rs]: typo: underlying

File: tokio/src/stream/skip.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+where
+    St: Stream,
+{
+    type Item = St::Item;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        loop {
+            match ready!(self.as_mut().project().stream.poll_next(cx)) {
+                Some(e) => {
+                    if *self.as_mut().project().remaining == 0 {
```
Comment (taiki-e) [inline tokio/src/stream/skip.rs]:
```suggestion
                    if self.remaining == 0 {
```

**Comment by Darksonn** [comment]
Did you mean to include that `skip_while.rs` file?

**Comment by torepettersen** [comment]
> Did you mean to include that `skip_while.rs` file?

No, sorry. I will be working on that next.


Also does anyone know why rustfmt is failing?

**Comment by Darksonn** [comment]
>Also does anyone know why rustfmt is failing?

It wants a semicolon on your return, and to remove the comma after the `}` in the match block.

**Pull request merged**